ol/tower/src/proof.rs

```diff
@@ -5,25 +5,27 @@ use crate::{backlog, delay::*, preimage::genesis_preimage};
 use anyhow::{bail, Error};
 use diem_global_constants::{genesis_delay_difficulty, GENESIS_VDF_SECURITY_PARAM};
 use diem_types::chain_id::NamedChain;
+use diem_logger::{info, debug, error};
 use glob::glob;
 use ol::node::client;
 use ol_types::block::VDFProof;
 use ol_types::config::AppCfg;
 use std::{fs, io::Write, path::PathBuf, time::Instant};
 use txs::tx_params::TxParams;
 
-/// name of the proof files
+/// The proofs filename
 pub const FILENAME: &str = "proof";
 
-// writes a JSON file with the first vdf proof
-fn mine_genesis(config: &AppCfg, difficulty: u64, security: u64) -> VDFProof {
-    println!("Mining Genesis Proof");
+/// Writes a JSON file with the first VDF proof
+fn mine_genesis(config: &AppCfg, difficulty: u64, security: u64) -> Result<VDFProof, Error> {
     let preimage = genesis_preimage(&config);
     let now = Instant::now();
 
-    let proof = do_delay(&preimage, difficulty, security).unwrap(); // Todo: make mine_genesis return a result.
+    info!("Mining genesis proof");
```

Comment:
Was unsure whether the `println` calls were deliberate or whether it was preferrable to use the `diem_logger`.